include/util.h

```diff
@@ -39,9 +39,24 @@ typedef enum {
 #define bcmp_topo_task_priority 3
 #endif
 
+typedef struct {
+  uint16_t year;
+  uint8_t month;
+  uint8_t day;
+  uint8_t hour;
+  uint8_t min;
+  uint8_t sec;
+  uint32_t usec;
+} UtcDateTime;
+
 #define array_size(x) (sizeof(x) / sizeof(x[0]))
 #define member_size(type, member) (sizeof(((type *)0)->member))
 
+uint32_t time_remaining(uint32_t start, uint32_t current, uint32_t timeout);
+uint32_t utc_from_date_time(uint16_t year, uint8_t month, uint8_t day,
+                            uint8_t hour, uint8_t minute, uint8_t second);
+void date_time_from_utc(uint64_t utc_us, UtcDateTime *date_time);
```

Comment:
Copied these over from app_util.h/util.c in bm_protocol + all of the help functions for leap years etc. I renamed them to match the code style guidelines.